assets/js/components/common/Debug.jsx

```diff
@@ -33,6 +38,21 @@ class Debug extends Component {
     }
   }
 
+  getSnapshotBeforeUpdate(prevProps, prevState) {
```

Comment:
does this function get called? how does this work

Reply:
Gets called before committing to the DOM and its result comes into `componentDidUpdate` through `snapshot`

https://reactjs.org/docs/react-component.html#getsnapshotbeforeupdate

Reply:
magic